ci-operator/step-registry/api-job/k8sgpt/api-job-k8sgpt-commands.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        # cat "${SHARED_DIR}/proxy-conf.sh"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+export KUBECONFIG=${KUBECONFIG}
+
+set_proxy
+run_command "oc whoami"
+run_command "oc version -o yaml"
+
+openai_token=$(cat "/var/run/vault/tests-private-account/openai-token")
+
+k8sgpt version
+set +x
+k8sgpt auth add -m gpt-3.5-turbo -b openai -p ${openai_token} | tee ${ARTIFACT_DIR}/k8sgpt-result || true
+set -x
+
+k8sgpt analyze -aed | tee -a ${ARTIFACT_DIR}/k8sgpt-result || true
+
+log=`cat ${ARTIFACT_DIR}/k8sgpt-result`
+
+if [[ ${log} =~ "No problems detected" ]]; then
+    cat >"${ARTIFACT_DIR}/k8sgpt-result.xml" <<EOF
```

Comment:
@jianzhangbjz just one suggestion so that https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-tools/job/prow-test-results-classfier/ could be easy to get result per current code:
put xml in junit, so the code could be
```shell
mkdir -p "${ARTIFACT_DIR}/junit"
if [[ ${log} =~ "No problems detected" ]]; then
    cat >"${ARTIFACT_DIR}/junit/k8sgpt-result.xml" <<EOF
    <testsuite name="api-job-k8sgpt" tests="1" failures="0">
        <testcase name="scanning cluster"/>
    </testsuite>
EOF
else
    cat >"${ARTIFACT_DIR}/junit/k8sgpt-result.xml" <<EOF
    <testsuite name="api-job-k8sgpt" tests="1" failures="1">
      <testcase name="scanning cluster">
        <failure message="">fail problems are detected</failure>
        <system-out>
          ${log}
        </system-out>
      </testcase>
    </testsuite>
EOF
fi

```